src/components/query-list/query-list.js

```diff
@@ -63,17 +100,26 @@ export default class QueryList extends Component {
       itemHeight,
       renderItem,
       length,
-      fullWidth
+      fullWidth,
+      page,
+      isMainPageSearch,
     } = this.props;
     const {
-      offset
+      offset,
     } = this.state;
 
+    const {
+      isPending,
+      length: totalResults,
+    } = collection;
+    const readOffset = isMainPageSearch ? page * PAGE_SIZE : offset;
```

Comment:
Maybe it would be better to split this on a few lines to increase readability?
```suggestion
    const readOffset = isMainPageSearch
      ? page * PAGE_SIZE
      : offset;
```

Reply:
I'm not sure about this string, because this string is not so long and values are simple for understanding. Line break is more appropriate when the values are complex

Reply:
Okay, I don't insist, let it be one line